leshan-core/src/main/java/org/eclipse/leshan/core/request/execute/Arguments.java

```diff
@@ -0,0 +1,105 @@
+package org.eclipse.leshan.core.request.execute;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+public class Arguments implements Iterable<SingleArgument> {
+
+    private final List<SingleArgument> argumentList;
+
+    public Arguments(List<SingleArgument> argumentList) {
+        this.argumentList = argumentList;
+    }
+
+    public Arguments(Map<Integer, String> argumentsMap) {
+        this(buildSingleArguments(argumentsMap));
+    }
```

Comment:
**maybe** we could add more convenient API like : 

Just some ideas : 
public Arguments(SingleArgument... argumentList);
public Arguments(String... values); // where digit in the index of the value.
public Arguments(Integer... digits); // for arguments without value only. 
public Arguments (Object... digitAndValues) where first element is first digit and second is first value and so on

Reply:
I'm not sure about public ```Arguments(String... values)``` and ```Arguments(Object... digitAndValues)``` constructors. In the first one there is not clear how to use it (without refering into javadoc) the string values ```"1", "2", "3", "4"``` could be interpreted as pair digit-values (1: "2", 3: "4") or just digits (1, 2, 3, 4). How about odd number of arguments? In the second one - we can pass list of any objects, and nothing prevent us against errors. The argument validation will be performed at runtime so it's harder to detect errors.

I am believer of preventing errors by compiler as much as possible (I think everyone hate runtime errors). Parsing and validating text into arguments is reasonable in one case - when we receive content from external world. In case invalid content we should throw some validation exception. But creating object programmatically from fuzzy arguments reminds me script programming and we reject the the great advantage of strong-typed language.

I know that some other objects are created in similar way, so it's argument to continue this for consistency. But how about introducing builder pattern here? It should be also convenient and it prevents against mistakes at compile time.

Reply:
No problem that was just idea, we will see later if we need some more constructors. 

> I know that some other objects are created in similar way, so it's argument to continue this for consistency. But how about introducing builder pattern here? It should be also convenient and it prevents against mistakes at compile time.

I don't maybe we should keep it simple and we will see if we need it at usage and so we could add it later ? 
(no strong opinion so if you feel we need it now, we can add it)

Reply:
I added builder for Argument and Arguments. If you don't like it, I'll remove it and add constructors like you proposed. If you accept this solution, I'll cleanup the code, add some javadoc and change draft to :slightly_smiling_face:

Reply:
I looked at this.
I understand the benefits about `Arguments` Builder but for `Argument` I'm not so sure ?

Reply:
You're right, I created Argument builder but it isn't used anywhere. Even if I replaced new Argument() with Argument.builder() it seems to not be helpful. So I removed it.

For sake of simplify I also removed other ways of creating Arguments than with builder. If you think that creating from map or argument list is useful, I'll restore it.